File: client-angular2/client/src/app/shared/model/objective.ts
import {ObjectiveAssociation} from './objective-association';

export interface Objective {
  id: number;
  companyId: number;
  title: string;
  description: string;
  percentage: number;
  estimatedCompletionDate: Date;
  objectiveAssociations: ObjectiveAssociation[];
}

File: server/Service/Objectives/Dto/Objective.cs
using Service.General.Dto;
using System;
using System.ComponentModel.DataAnnotations;

namespace Service.Objectives.Dto
{
    public abstract class Objective : BaseDto, IDto
    {
        public int Id { get; set; }

        [MaxLength(100)]
        public string Title { get; set; }

        public string Description { get; set; }

        public decimal Percentage { get; set; }

        public DateTime EstimatedCompletionDate { get; set; }

        public KeyResult[] KeyResults { get; set; }

        public ObjectiveAssociation[] ObjectiveAssociations { get; set; }
    }
}